Microsoft.Toolkit.Uwp.UI/Media/CanvasBrushBase.cs

```diff
@@ -38,12 +38,24 @@ public abstract class CanvasBrushBase : XamlCompositionBrushBase
         /// <returns>True if drawing was completed and the brush is ready, otherwise return False to not create brush yet.</returns>
         protected abstract bool OnDraw(CanvasDevice device, CanvasDrawingSession session, Vector2 size);
 
+        /// <summary>
+        /// Gets or sets the shared device.
+        /// </summary>
+        protected CanvasDevice Device { get; set; }
+
+        /// <summary>
+        /// Gets or sets the internal graphics composition.
+        /// </summary>
+        protected CompositionGraphicsDevice Graphics { get; set; }
+
         /// <summary>
         /// Initializes the Composition Brush.
         /// </summary>
         protected override void OnConnected()
         {
             base.OnConnected();
+            Device = CanvasDevice.GetSharedDevice();
+            Graphics = CanvasComposition.CreateCompositionGraphicsDevice(Window.Current.Compositor, Device);
 
             // Delay creating composition resources until they're required.
             if (CompositionBrush == null)
```

Comment:
I think these checks are still important no?  Especially the one about effects being supported?